tools/ci/java-ci-tools/src/main/java/org/apache/flink/tools/ci/licensecheck/LicenseChecker.java

```diff
@@ -0,0 +1,320 @@
+		File moduleFile = noticeFile.getParentFile() // META-INF
+			.getParentFile() // resources
+			.getParentFile() // main
+			.getParentFile() // src
+			.getParentFile(); // <-- module name
+		return moduleFile.getName();
+	}
+
+	private int checkNoticeFile(Multimap<String, IncludedDependency> modulesWithShadedDependencies, File noticeFile) throws IOException {
+		int severeIssueCount = 0;
+		String moduleName = getModuleFromNoticeFile(noticeFile);
+
+		// 1st line contains module name
+		String noticeContents = readFile(noticeFile.toPath());
+		if (!noticeContents.startsWith(moduleName)) {
+			String firstLine = noticeContents.substring(0, noticeContents.indexOf('\n'));
+			LOG.warn("Expected first file of notice file to start with module name. moduleName={}, firstLine={}", moduleName, firstLine);
+		}
+
+		// collect all declared dependencies from NOTICE file
+		Set<IncludedDependency> declaredDependencies = new HashSet<>();
+		try (BufferedReader br = new BufferedReader(new StringReader(noticeContents))) {
+			String line;
+			while ((line = br.readLine()) != null) {
+				Matcher noticeDependencyMatcher = NOTICE_DEPENDENCY_PATTERN.matcher(line);
+				if (noticeDependencyMatcher.find()) {
+					String groupId = noticeDependencyMatcher.group(1);
+					String artifactId = noticeDependencyMatcher.group(2);
+					String version = noticeDependencyMatcher.group(3);
+					if (groupId == null && artifactId == null && version == null) { // "bundles" case
+						groupId = noticeDependencyMatcher.group(5);
+						artifactId = noticeDependencyMatcher.group(6);
+						version = noticeDependencyMatcher.group(7);
+					}
+					IncludedDependency toAdd = IncludedDependency.create(groupId, artifactId, version);
+					if (!declaredDependencies.add(toAdd)) {
+						LOG.warn("Dependency {} has been declared twice in module {}", toAdd, moduleName);
+					}
+				}
+			}
+		}
+		// print all dependencies missing from NOTICE file
+		Set<IncludedDependency> expectedDependencies = new HashSet<>(modulesWithShadedDependencies.get(moduleName));
+		expectedDependencies.removeAll(declaredDependencies);
+		for (IncludedDependency missingDependency : expectedDependencies) {
+			LOG.error("Could not find dependency {} in NOTICE file {}", missingDependency, noticeFile);
+			severeIssueCount++;
+		}
+
+		if (!MODULES_DEFINING_EXCESS_DEPENDENCIES.contains(moduleName)) {
+			// print all dependencies defined in NOTICE file, which were not expected
+			Set<IncludedDependency> excessDependencies = new HashSet<>(declaredDependencies);
+			excessDependencies.removeAll(modulesWithShadedDependencies.get(moduleName));
+			for (IncludedDependency excessDependency : excessDependencies) {
+				LOG.warn("Dependency {} is mentioned in NOTICE file {}, but is not expected there", excessDependency, noticeFile);
+			}
+		}
+
+		return severeIssueCount;
+	}
+
+	private static String readFile(Path path) throws IOException {
+		byte[] encoded = Files.readAllBytes(path);
```

Comment:
any particular reason you are reading everything into a single string, instead of using `Files.readAllLines()`?